src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs

```diff
@@ -103,7 +105,7 @@ protected override void OnContentChanged(object oldContent, object newContent)
 
 		if (oldContent is FrameworkElement oldElement)
 		{
-			_canvas?.Children.Remove(oldElement);
+			_owner?.Children.Remove(oldElement);
 			oldElement.SizeChanged -= OnContentSizeChanged;
 		}
```

Comment:
We should not use this approach to invalidate the layout when the size changed because it will requeue a MEASURE_DIRTY (invalidate measure) on the scheduler and schedule a layout recalculation.

Instead, we should use the ArrangeOverride() opportunity to adjust the size of the shadow container to the right size.